hasura.planx.uk/migrations/1737881459011_create_view_email_events_log/up.sql

```diff
@@ -0,0 +1,52 @@
+CREATE OR REPLACE VIEW public.email_events_log AS 
+WITH retries AS (
+   SELECT hdb_scheduled_event_invocation_logs.id
+   FROM hdb_catalog.hdb_scheduled_event_invocation_logs
+   WHERE ((hdb_scheduled_event_invocation_logs.event_id, hdb_scheduled_event_invocation_logs.created_at) IN ( 
+      SELECT 
+         seil.event_id,
+         max(seil.created_at) AS max
+      FROM (hdb_catalog.hdb_scheduled_event_invocation_logs seil
+      LEFT JOIN hdb_catalog.hdb_scheduled_events se ON ((se.id = seil.event_id)))
+      WHERE (se.tries > 1)
+      GROUP BY seil.event_id))
+), emails AS (
+   SELECT 
+      ((((seil.request -> 'payload'::text) -> 'payload'::text) ->> 'sessionId'::text))::uuid AS session_id,
+      se.id AS event_id,
+   CASE
+      WHEN ((webhook_conf)::text ~~ '%/send-email/expiry"'::text) THEN 'Session expiry'::text
+      WHEN ((webhook_conf)::text ~~ '%/send-email/reminder"'::text) THEN 'Session reminder'::text
+      WHEN ((webhook_conf)::text ~~ '%/send-email/confirmation"'::text) THEN 'Submission confirmation'::text
+      WHEN ((webhook_conf)::text ~~ '%/send-email/invite-to-pay"'::text) THEN 'Invitation to pay'::text
+      WHEN ((webhook_conf)::text ~~ '%/send-email/invite-to-pay-agent"'::text) THEN 'Invitation to pay - agent'::text
+      WHEN ((webhook_conf)::text ~~ '%/send-email/payment-expiry"'::text) THEN 'Payment expiry'::text
+      WHEN ((webhook_conf)::text ~~ '%/send-email/payment-expiry-agent"'::text) THEN 'Payment expiry - agent'::text
+      WHEN ((webhook_conf)::text ~~ '%/send-email/payment-reminder"'::text) THEN 'Payment reminder'::text
+      WHEN ((webhook_conf)::text ~~ '%/send-email/payment-reminder-agent"'::text) THEN 'Payment reminder - agent'::text
+      ELSE (webhook_conf)::text
```

Comment:
Great fallback to catch new templates or other issues 👍 

The fact we have to parse this from here (but it's very useful information) suggests we should maybe consider capturing this more explicitly / pro-actively as an audit log when sending emails.

This is a really solid and practical solution though - love it 👍